bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,171 @@
+        keepAliveTimeout = (Integer) parameters.get(CONFIG_KEEP_ALIVE);
+        queueSize = (Integer) parameters.get(CONFIG_QUEUE_SIZE);
+        threadPool = new ThreadPoolExecutor(minThreads, maxThreads, keepAliveTimeout, TimeUnit.SECONDS,
+                new ArrayBlockingQueue<Runnable>(queueSize), new NamedThreadFactory("jetty"));
+        logger.info("jetty thread pool started with min threads {}, max threads {}", minThreads, maxThreads);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> parameters) {
+        minThreads = (Integer) parameters.get(CONFIG_MIN_THREADS);
+        maxThreads = (Integer) parameters.get(CONFIG_MAX_THREADS);
+        keepAliveTimeout = (Integer) parameters.get(CONFIG_KEEP_ALIVE);
+        queueSize = (Integer) parameters.get(CONFIG_QUEUE_SIZE);
+        threadPool.setCorePoolSize(minThreads);
+        threadPool.setMaximumPoolSize(maxThreads);
+        threadPool.setKeepAliveTime(keepAliveTimeout, TimeUnit.SECONDS);
+        logger.debug("jetty thread pool changed to min threads {}, max threads {}", minThreads, maxThreads);
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        threadPool.shutdown();
+        logger.info("jetty thread pool shutdown");
+    }
+
+    @Override
+    public HttpClient getHttpClient(String consumerName, String endpoint) {
+        Objects.requireNonNull(endpoint, "endpoint must not be null");
+        logger.debug("httpClient for endpoint {} requested", endpoint);
+        checkConsumerName(consumerName);
+        return createHttpClient(consumerName, endpoint);
+    }
+
+    @Override
+    public HttpClient getHttpClient(String consumerName) {
+        logger.debug("httpClient for requested");
+        checkConsumerName(consumerName);
+        return createHttpClient(consumerName, null);
+    }
+
+    private HttpClient createHttpClient(String consumerName, String endpoint) {
+        logger.info("creating httpClient for endpoint {}", endpoint);
+        SslContextFactory sslContextFactory = new SslContextFactory();
+        sslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
+        if (endpoint != null && customTrustmanagerProvider != null) {
+            Stream<TrustManager> trustManagerStream = customTrustmanagerProvider.getTrustManagers(endpoint);
+            TrustManager[] trustManagers = trustManagerStream.toArray(TrustManager[]::new);
+            if (trustManagers.length > 0) {
+                logger.info("using custom trustmanagers (certificate pinning) for httpClient for endpoint {}",
+                        endpoint);
+                try {
+                    SSLContext sslContext = SSLContext.getInstance("TLS");
+                    sslContext.init(null, trustManagers, null);
+                    sslContextFactory.setSslContext(sslContext);
+                } catch (Exception ex) {
+                    throw new RuntimeException("Cannot create an TLS context for the endpoint '" + endpoint + "'!", ex);
```

Comment:
This runtime exception needs to be documented in javadoc of `HttpClientFactory#getHttpClient(String)`